src/viperleed/calc/files/beams.py

```diff
@@ -239,11 +241,12 @@ def readOUTBEAMS(filename="EXPBEAMS.csv", sep=",", enrange=None):
             with open(filename, 'r') as rf:
                 lines = [li[:-1] for li in rf.readlines()]
         except FileNotFoundError:
-            if filename.endswith(".csv") and os.path.isfile(filename[:-4]):
-                with open(filename[:-4], 'r') as rf:
+            filename = Path(filename)
+            if filename.suffix == '.csv' and filename.with_suffix('').is_file:
```

Comment:
```suggestion
            if filename.with_suffix('').is_file():
```

We probably do not need to check that it was a `.csv`. The earlier check was because of the `[:-4]` slice, I guess. Notice also that you did not call `.is_file()` but would be checking the existence of the method.